### DISCUSSION TITLE
FP16 optimization on CPU

### DISCUSSION THREAD

**Discussion by jeyblu** [microsoft/onnxruntime#18701] [Other Q&A] [answered]
Are FP16 code optimized with AVX or other vector instructions for CPU? Thanks.

[MlasHalfGemmKernel<MLAS_HALF_GEMM_KERNEL_DEFAULT>](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/mlas/lib/halfgemm.cpp#L291) doesn't seem to have any vector instructions.

**Comment by chenfucn** [comment, marked as answer]
your understanding is correct. it is not optimized for avx. We have optimized kernel for ARM neon chips and in the process of refining that:
https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/mlas/lib/halfgemm_kernel_neon.cpp

**Reply by jeyblu** [reply]
Are customers asking for FP16 support on ARM neon chips and not on Intel/AMD chips? Thanks

**Comment by deischi** [comment]
The OpenVINO execution provider seems to have FP16 support for CPU (CPU_FP16)
https://onnxruntime.ai/docs/execution-providers/OpenVINO-ExecutionProvider.html#summary-of-options

But honestly I do not know details about that.
